assets/js/datagovsg-search.js

```diff
@@ -66,25 +77,48 @@ function databaseSearch(searchTerm, index, searchField) {
   var request = $.ajax({
     url: 'https://data.gov.sg/api/action/datastore_search',
     data: data,
-    dataType: 'json'
+    dataType: 'json',
+    success: callback
   });
 
   request.done(function (data) {
-    document.getElementById("loading-spinner").style.display = 'none';
-    hideAllPostsAndPagination();
+    datagovsgTotal = data.result.total;
+    if (isFirstRender) {
+      pageResults = Array(Math.ceil(datagovsgTotal / PAGINATION_DISPLAY_RESULTS_PER_PAGE)).fill(null);
```

Comment:
hmm why this arh? shouldnt we be loading them until the info is back rather than displaying a false null?

Reply:
This is a quirk of how the view is being generated - whenever a page loads new data, we actually replace the html in the component corresponding to that page number, while hiding everything out of range. In practice users won't get to see blank results because there's a bit of lead time where each section is preloaded to replace the null values several pages before/after